subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/JfrProfiler.groovy

```diff
@@ -98,6 +98,13 @@ class JfrProfiler extends Profiler implements Stoppable {
     }
 
     void start(BuildExperimentSpec spec) {
+        // Remove any profiles created during warmup
```

Comment:
❌ This shouldn't be necessary - When using no-daemon, we should only be passing the profiling arguments to the builds we want to profile.

Reply:
That would be nicer I agree. IIUC, the code currently assumes that the command-line is going to be the same for every invocation. Changing that in a nice way would require some non-trivial refactoring, which I won't have time for now. I'm happy to create an issue for it, though.

Meanwhile, I think this change is very useful as-is, and would still like to merge it. I can leave a TODO pointing to the issue if you want.

Reply:
Yes, please leave a TODO, so if someone stumbles over this they know how to improve it.

Reply:
Done.